### DISCUSSION TITLE
Implementation of the scikit-image Rolling Ball Algorithm

### DISCUSSION THREAD

**Discussion by dxvidparham** [clEsperanto/pyclesperanto#147] [Ideas]
Are there any ideas of adding the rolling ball algorithm for backgroung removal, as an alternative to the top_hat approach?

**Comment by StRigaud** [comment]
Hi @davidlbit

Thanks for you question!

The filter was not explicitly requested before, so we haven't look seriously on implementing it inside clesperanto although I belive it can be. But from my experience, the Top Hat provide a very similar results than the Rolling Ball for much less ressources (memory and speed) and complexity. 

Is there any particular advantage for using the Rolling Ball?

PS: I assume you meant scikit-image :wink:

**Reply by dxvidparham** [reply]
Yes, I ment to say scikit-image :D

**Comment by haesleinhuepf** [comment]
Hi @davidlbit ,

I was thinking of implementing it on the GPU; and at some point there was actually a community challenge in this context. [In the related discussion](https://forum.image.sc/t/rolling-ball-background-subtraction-challenge/31837/14?u=haesleinhuepf), the original developer of the rolling-ball algorithm in ImageJ mentioned that he often uses top-hat nowadays. This was for me a reason to not dive deeper into the topic.

Curious: Can you give an example data set where the rolling-ball algorithm does a good job and the top-hat filter does not?

Thanks!

Best,
Robert

**Comment by dxvidparham** [comment]
Hey @haesleinhuepf and @StRigaud,

Thanks for getting back to me so quickly! I've noticed that the rolling ball algorithm can be pretty demanding computationally, which got me thinking about other options. I remember stumbling upon some old forum posts that praised the rolling ball method, which might be why I haven't explored the top-hat approach much. But after hearing your thoughts, I'm definitely going to give it another look.

**Reply by StRigaud** [reply]
We are interested if you found that the Rolling Ball is still worth it in terms of results. Don't hesitate to share your conclusion!
